### PR TITLE
[Enhance](auth)Users support multiple roles

### PR DESCRIPTION
**Pull request by zddr**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.
1.support GRANT role [, role] TO user_identity
2.support REVOKE role [, role] FROM user_identity
3.’Show grants‘ Add a column to display the roles owned by users
4.‘alter user’ prohibit deleting user's role
5.Repair Logic of roleName cannot start with RoleManager.DEFAULT_ ROLE

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run build all

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/AuthProcDir.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
  */
 public class AuthProcDir implements ProcDirInterface {
     public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
-            .add("UserIdentity").add("Password").add("GlobalPrivs").add("CatalogPrivs")
+            .add("UserIdentity").add("Password").add("Roles").add("GlobalPrivs").add("CatalogPrivs")
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/AuthProcDir.java:34]: Is it suitable to add a new column here?
how about move it to `show roles`?

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/Auth.java [hunk 2]
```diff
@@ -668,6 +701,30 @@ private void revokeInternal(UserIdentity userIdent, String role, ResourcePattern
...
+            if (userManager.getUserByUserIdentity(userIdent) == null) {
+                throw new DdlException("user: " + userIdent + " does not exist");
+            }
+            //roles must exist
+            for (String roleName : roles) {
+                if (roleManager.getRole(roleName) == null) {
+                    throw new DdlException("role:" + roleName + " does not exist");
+                }
+            }
+            userRoleManager.reduceUserRole(userIdent, roles);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/Auth.java:717]:
```suggestion
            userRoleManager.removeUserRole(userIdent, roles);
```
add <-> remove

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserRoleManager.java [hunk 3]
```diff
@@ -58,6 +61,35 @@ public void addUserRole(UserIdentity userIdentity, String roleName) {
         roleToUsers.put(roleName, userIdentities);
     }
 
+    public void addUserRole(UserIdentity userIdentity, List<String> roles) {
+        for (String roleName : roles) {
+            addUserRole(userIdentity, roleName);
+        }
+    }
+
+    public void reduceUserRole(UserIdentity userIdentity, List<String> roles) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/UserRoleManager.java:70]:
```suggestion
    public void reduceUserRoles(UserIdentity userIdentity, List<String> roles) {
```

File: regression-test/suites/account_p0/test_alter_user.groovy [hunk 4]
```diff
@@ -17,32 +17,9 @@
...
-
-    sql """grant load_priv on ctl.*.* to test_auth_user1"""
-    sql """grant load_priv on ctl.*.* to role 'test_auth_role2'"""
-
-    // change user's role again
-    sql """alter user test_auth_user1 default role 'test_auth_role1'"""
-    order_qt_show_grants3 """show grants for 'test_auth_user1'"""
     
     // 2. test password history
     sql """set global password_history=0""" // disabled
```
Comment (morningman) [inline regression-test/suites/account_p0/test_alter_user.groovy:25]: Need to add case for new feature

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/AuthProcDir.java (hunk 1 above)
Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/common/proc/AuthProcDir.java:34]: Show roles is to show which users have roles assigned to them. Here is to show which roles users have

File: regression-test/suites/account_p0/test_alter_user.groovy (hunk 4 above)
Comment (zddr) [reply regression-test/suites/account_p0/test_alter_user.groovy:25]: Corresponding use cases have been added to fe ut

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**